clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -92,6 +93,15 @@ public MetadataRequest build(short version) {
             if (!data.allowAutoTopicCreation() && version < 4)
                 throw new UnsupportedVersionException("MetadataRequest versions older than 4 don't support the " +
                         "allowAutoTopicCreation field");
+            if (version >= 10) {
+                if (data.topics() != null) {
+                    data.topics().forEach(topic -> {
+                        if (topic.name() == null || topic.topicId() != Uuid.ZERO_UUID)
+                            throw new UnsupportedVersionException("MetadataRequest version " + version  +
+                                    " does not support null topic names or topic IDs.");
```

Comment:
Slight clarification: `or non-null topic IDs`

Reply:
Would it make sense to say non-zero topic IDs? Since the null ID is represented with all zeros?

Reply:
Yeah, makes sense.